src/components/RegionSelect/RegionSelect.tsx

```diff
@@ -1,68 +1,130 @@
+  { groupKey: 'countries_with_coral', groupName: 'Countries with Coral Reefs', label: 'Jamaica' },
+  { groupKey: 'countries_with_coral', groupName: 'Countries with Coral Reefs', label: 'Malaysia' },
+  {
+    groupKey: 'countries_with_coral',
+    groupName: 'Countries with Coral Reefs',
+    label: 'New Zealand',
+  },
+  {
+    groupKey: 'countries_with_coral',
+    groupName: 'Countries with Coral Reefs',
+    label: 'Saint Kitts',
+  },
+  { groupKey: 'coral_reef_regions', groupName: 'Coral reef regions', label: 'Great Barrier Reef' },
+  { groupKey: 'coral_reef_regions', groupName: 'Coral reef regions', label: 'Caribbean Sea' },
+  { groupKey: 'coral_reef_regions', groupName: 'Coral reef regions', label: 'Red Sea' },
+  { groupKey: 'coral_reef_regions', groupName: 'Coral reef regions', label: 'Indo-Pacific' },
+  { groupKey: 'coral_reef_regions', groupName: 'Coral reef regions', label: 'Coral Triangle' },
+  { groupKey: 'coral_reef_regions', groupName: 'Coral reef regions', label: 'Mesoamerican Reef' },
+  {
+    groupKey: 'coral_reef_regions',
+    groupName: 'Coral reef regions',
+    label: 'Hawaiian Archipelago',
+  },
+  { groupKey: 'coral_reef_regions', groupName: 'Coral reef regions', label: 'Ningaloo Reef' },
 ]
 /** End of mock data */
 
 export default function RegionSelect() {
   const { t } = useTranslation()
-  const [selectedValue, setSelectedValue] = useState<CountryOption | null>(DEFAULT_OPTION)
+  const [selectedValue, setSelectedValue] = useState<RegionOption | null>(defaultOption)
+  const noCountriesMatchText = t('no_countries_match')
+  const noRegionsMatchText = t('no_regions_match')
 
-  const sortedOptions = countryOptions.sort((a, b) => {
-    if (a.group !== b.group) {
-      return a.group.localeCompare(b.group)
-    }
-    return a.label.localeCompare(b.label)
-  })
+  const sortedOptions = useMemo(() => {
+    return [...regionOptions].sort((a, b) => {
+      if (a.groupKey !== b.groupKey) {
+        const aIndex = groupOrders.indexOf(a.groupKey)
+        const bIndex = groupOrders.indexOf(b.groupKey)
+        return aIndex - bIndex
+      }
+      return a.label.localeCompare(b.label)
+    })
+  }, [])
 
-  const handleChange = (_, newValue: CountryOption | null) => {
-    setSelectedValue(newValue || DEFAULT_OPTION)
-  }
+  const muiFilterOptions = createFilterOptions<RegionOption>({ ignoreAccents: true, trim: true })
 
-  const isOptionEqual = (option: CountryOption, value: CountryOption) => {
-    return option.label === value.label && option.group === value.group
+  const createEmptyFilterOptions = useCallback((): RegionOption[] => {
+    const noDataGroups = [
+      {
+        groupKey: 'countries_with_coral',
+        groupName: 'Countries with Coral Reefs',
+        label: noCountriesMatchText,
+      },
+      {
+        groupKey: 'coral_reef_regions',
+        groupName: 'Coral reef regions',
+        label: noRegionsMatchText,
+      },
+    ]
+
+    return noDataGroups as RegionOption[]
```

Comment:
Non-blocking suggestion: 
```suggestion
    return [
      {
        groupKey: 'countries_with_coral',
        groupName: 'Countries with Coral Reefs',
        label: noCountriesMatchText,
      },
      {
        groupKey: 'coral_reef_regions',
        groupName: 'Coral reef regions',
        label: noRegionsMatchText,
      },
    ]
```